### DISCUSSION TITLE
Can not find proper return type for Joins in stored procedure

### DISCUSSION THREAD

**Discussion by Qura-tul-ain** [supabase/supabase#13666] [Questions] [answered]
Hi, i have written a stored procedure to do text search in a table named "jobs".

Now i want to add a join query in it but can not find a proper return type.

Here is my code

create or replace function primary_search(text_input text, region text)
returns setof jobs
language sql
as $$
  select *
  from jobs
  join employers on jobs."employerId" = employers.id 
  and to_tsvector(categories) @@ to_tsquery(text_input) and to_tsvector(location) @@ to_tsquery(region) ';
$$;

it was working fine before Joining because it is returning result from a single table. 
Now im getting this error:
Failed to run sql query: return type mismatch in function declared to return jobs

Can anyone help me with this, thanks.

**Comment by steve-chavez** [comment, marked as answer]
In this case you'll have to use a `RETURNS TABLE(col1 type, col2 type, ..)` and list all the columns types you want to return.
